app/code/Magento/Catalog/Model/Product/Image/Process/Cache.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Catalog\Model\Product\Image\Process;
+
+class Cache extends \Magento\Catalog\Model\Product\Image\Cache
```

Comment:
Magento does not recognize inheritance as a good practice at the moment. 
Please look into the possibility to extract required business logic into the different class and refactor the old code to use it as well.